public/pages/Rules/components/RuleViewerFlyout/RuleViewerFlyout.tsx

```diff
@@ -74,6 +81,9 @@ export const RuleViewerFlyout: React.FC<RuleViewerFlyoutProps> = ({
   };
 
   const onDeleteRuleConfirmed = async () => {
+    if (!ruleService) {
+      return;
+    }
     const deleteRuleRes = await ruleService.deleteRule(ruleTableItem.ruleId);
 
     if (!deleteRuleRes.ok) {
```

Comment:
Let's fix this Todo? we can show an error toast saying `Failed to delete rule`

Reply:
check this commit, I think its a right error handling here now:
https://github.com/opensearch-project/security-analytics-dashboards-plugin/pull/219/commits/dae1c8f4242676415e56b16a8999d9d85e5764d1#diff-ddf4d433db9a97c023d49ad7aeb681cf5376c5248d42f51246485942e687863fR86